dean.js

```diff
@@ -0,0 +1,136 @@
+/** @param {NS} ns **/
+/*
+create list of all servers
+iterate through list:
+count programs
+open ports
+nuke
+scp and exec hack script
+if all 5 ports are open then everything above has run, remove from list
+*/
+export async function main(ns) {
+	//create array of all servers except purchased called listServers[]
+	var listServers = await ns.scan('home');
+	let listIndex = 0;
+
+	while (listIndex < listServers.length) {
+		await ns.sleep(50);
+		let listScan = ns.scan(listServers[listIndex], true);
+		for (let i = 0; i < listScan.length; i++) {
+			if (listServers.indexOf(listScan[i]) === -1) {
```

Comment:
`.includes()` is a newer JS function, but simply returns true/false if item is in the array. A common bug is that people will check if the result of `indexOf` returns false if not found, but it's `-1`.

```suggestion
			if (listServers.includes(listScan[i])) {
```

Reply:
done but with **!**listServers.includes(listScan[i])